### ISSUE TITLE
A held light source in multiplayer does not appear for other players

### ISSUE THREAD

**Issue by Cervator** [MovingBlocks/Terasology#3280]
### What you were trying to do

Test unrelated stuff in multiplayer. Since the game started in the dark I had both connected players (to a local headless server) hold torches to see things. I was expecting player 1 would then have an easy time seeing player 2 with a held torch

### What actually happened

The held torch for the player *other* than the local client did not give off any light. A *placed* torch from the other player would light up fine, just a held torch would only yield light on the local client.

Same situation with a server/client + client only

### How to reproduce

* Run two instances of the game
* Join together in multiplayer
* Note the absence of light in the environment when the *other* player is holding a torch

### Theory

The logic behind held items have been a hot topic for some time with regular changes, especially related to entity replication over the network in multiplayer. Since the held item on a client is only visible on that client probably at some point that was made *too* local and no longer replicates fully if at all. Probably this isn't actually rendering related, the light isn't being rendered because the engine doesn't know it is supposed to be there (for other clients)

This is also wrong in another way as past proof of concept code (see #2115 and #2133) *needs* some entity information that might at first glance seem local, but could be used to visually display extras on another visible player in multiplayer, like posing the head of the model in the direction the player is facing or indeed show the item they are wielding attached to their model

Search for classes with the word `Held` in the name for the related code and consider doing the same in the issue tracker, especially for closed issues, to see what other changes have been made over time :-) Be especially mindful of having a player log off and back on repeatedly to verify all is still well, and check out location 0,0,0 in the world as past held item issues have resulted in an orphan item appearing there. Test both headless + 2 players and hosting player + client player.

See https://metaterasology.github.io/docs/developing/networkMultiplayer/networkMultiplayer.html for some related documentation

While I think this could be a good first issue in some cases be warned that it may take a bit more research and testing than most beginner friendly issues, and a computer able to at least run two game clients at the same time :-)

Edit: Screenshot

![heldlightissueinmultiplayer](https://user-images.githubusercontent.com/1063833/36930261-1fdee62a-1e6b-11e8-8671-7b4063856bcc.png)

**Comment by dorasun** [comment]
A friend and I tested this on the unstable multiplayer server (we were both testing on MacOS, 10.11.6 and 10.12.5) and we noticed that, as you mentioned, you can't see any items anyone else is holding. We took a look at the two files with the word 'held' in them (ChangeHeldItemRequest.java and BuilderHeldItemAuthoritySystem.java), as you suggested, but aren't seeing any way to edit the actual models. If you don't mind, could you point us in the direction of the user model? I did a search on IntelliJ and got too many results for just 'model', and I'm not sure what the correct term is.

**Comment by dave2s** [comment]
Hello everyone,

I am happy to announce that I've made some progress regarding this issue. I have a created a proof of concept fix for this issue, which includes remote players held items as well. I'll be creating a PR in the upcoming weeks, so If any one of you is still trying to wrap your head around this, it might be best to wait.
The way I went about this is I took @Josharias 's firstPersonClientSystem concept and forked it so it takes care of other players in a multiplayer session. This includes remote player mountpoint and transform components with the difference of mountpoints being attached to remote character instead of local camera. Every client then has a remote player system which handles all other clients in a multiplayer session, attaches mountpoints to them, attaches held items to the mountpoints and takes care of the transformations.
